src/Sylius/Component/Promotion/Model/CatalogPromotionRuleInterface.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Promotion\Model;
+
+use Sylius\Component\Resource\Model\ResourceInterface;
+
+interface CatalogPromotionRuleInterface extends ResourceInterface
+{
+    public const TYPE_CONTAINS_VARIANTS = 'contains_variants';
```

Comment:
IMO this const should be defined in Core as Promotion component has no dependency to Product bundle/component

Reply:
yup, it was added in https://github.com/Sylius/Sylius/pull/13029